File: myexamples/imProcessingBench/bench_gaussian.cpp
//Apply Gaussian Fillter

#include <iostream>
#include <vector>
#include <chrono>

#include "opencv2/imgproc/imgproc.hpp"
#include "opencv2/highgui/highgui.hpp"


#define NTIMES 1000
#define KERNEL_LENGTH 5

using namespace std;
using namespace cv;

int main(int argc, char* argv[]){
	Mat img;
	double proccessingTime=0;
	if(argc!=2){
		cerr<<"usage filename"<<endl;
		exit(1);
	}
	img=imread(argv[1]);
	cout<<img<<endl;
	cerr<<"ch,dim"<<img.channels()<<", "<<img.dims<<std::endl;
	for(int i=0;i<NTIMES;i++){
		auto start=std::chrono::system_clock::now();
		GaussianBlur( img, img, Size( KERNEL_LENGTH, KERNEL_LENGTH ),0);
		auto end=std::chrono::system_clock::now();
		proccessingTime+=std::chrono::duration_cast<std::chrono::milliseconds>(end-start).count();
	}
	imwrite("cfilterd.jpg",img);
	cerr<<"filter end"<<endl;
	cout<<"Time "<<proccessingTime<<"ms"<<std::endl;
	return 0;
}

File: myexamples/cvBinSendSample/bench_gaussian.py
# Apply Gaussian Fillter

import sys
#import numpy as np
import cv2
import time

NTIMES=100

if __name__ == '__main__':
	if len(sys.argv)!=2:
		print "usage filename"
		sys.exit(1)
	img = cv2.imread(sys.argv[1])
	processingTime=0
	for i in range(0,NTIMES):
		#sttime=time.time()
		retImg =cv2.GaussianBlur(img,(5,5),0)
		#edtime=time.time()
		#processingTime+=(edtime-sttime)
	cv2.imwrite("filterd.jpg",retImg)
	print "filter end"
	print "Time "+str(processingTime)+"ms"
